mne/encoding/model.py

```diff
@@ -0,0 +1,230 @@
+"""Base and high-level classes for fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from ..externals.six import string_types
+
+
+class SampleMasker(object):
+    """Remove samples with a particular value before fitting.
+
+    Parameters
+    ----------
+    estimator : instance of sklearn-style estimator
+        The estimator to be called after samples are removed.
+    mask_val : float, int, nan | callable.
```

Comment:
here the `ixs` and `mask_val` parameters do two different things. If `ixs` is not None, then it will act as a first-pass mask. If `mask_val` is not None, then it will remove rows where it finds that value. I'll rename `ixs` to `train_samples` etc as per your subsequent comments